lib/TransactionRunner.js

```diff
@@ -572,16 +572,20 @@ include a message body: https://tools.ietf.org/html/rfc7231#section-6.3\
         // Create test message from messages of all validation errors
         let message = '';
         const object = gavelResult || {};
+
         let validatorOutput;
-        for (const sectionName of Object.keys(object || {})) {
-          // Section names are 'statusCode', 'headers', 'body' (and 'version', which is irrelevant)
+
+        // Order-sensitive list of validation sections to output in the log
+        const resultSections = ['headers', 'body', 'statusCode']
+          .filter(sectionName => Object.prototype.hasOwnProperty.call(gavelResult, sectionName));
```

Comment:
Since now we have an explicit list of sections we need to filter out those sections that are actually present in the Gavel's results.